commands/command_filter.go

```diff
@@ -20,6 +20,37 @@ const (
 	SmudgeOperation
 )
 
+type InputFileHdr struct {
+	FileName string
+	FileLen  uint32
+}
+
+func (h *InputFileHdr) Read(r io.Reader) error {
+	// Read file name length
+	var fileNameLen uint32
+	if err := binary.Read(r, binary.LittleEndian, &fileNameLen); err != nil {
+		return errutil.Errorf(err, "Error reading filename length.")
+	}
+
+	// Read file name
+	h.FileName = ""
+	if fileNameLen > 0 {
+		buf := make([]byte, fileNameLen)
+		readLen, err := r.Read(buf)
+		if err != nil || readLen != int(fileNameLen) {
```

Comment:
Hmm, I would split out these two cases individually. Returning the error if it's `!= nil` would be the first, and the unexpected EOF would be the second. This makes things a little clearer since we can get an "error" when the read succeeded, just read less than we wanted. I'm thinking:

```go
if readLen, err := r.Read(buf); err != nil {
        return errutil.Errorf(err, "Unexpected error")
} else if readLEn != int(fileNameLen) {
        return fmt.Errorf("unexpected EOF when reading file (got %d, wanted %d)", readLen, fileNameLen)
}
```